lib/process-manager.ts

```diff
@@ -557,6 +557,12 @@ export class QueryProcessManager<T = string, U = string> extends ProcessManager<
 			process.exit();
 		});
 	}
+	destroyProcess(process: QueryProcessWrapper<T, U>) {
+		process.destroy();
+		process.process.removeAllListeners('disconnect');
+		process.process.kill();
+		this.processes.splice(this.processes.indexOf(process), 1);
```

Comment:
Should probably make sure that `this.processes.indexOf(process)` doesn't return `-1`.